pkg/probe/probes.go

```diff
@@ -77,174 +77,165 @@ func currentStateAsGJson() (gjson.Result, error) {
-		if defaultGw == "" {
-			msg := "default gw missing"
-			defaultGwLog := log.WithValues("path", defaultGwGjsonPath)
-			defaultGwLogDebug := defaultGwLog.V(1)
-			if defaultGwLogDebug.Enabled() {
-				defaultGwLogDebug.Info(msg, "state", gjsonCurrentState.String())
-			} else {
-				defaultGwLog.Info(msg)
-			}
-			return false, nil
+	gjsonCurrentState, err := currentStateAsGJson()
+	if err != nil {
+		return "", errors.Wrap(err, "failed retrieving current state to retrieve default gw")
+	}
+	defaultGwGjsonPath := "routes.running.#(destination==\"0.0.0.0/0\").next-hop-address"
+	defaultGw := gjsonCurrentState.Get(defaultGwGjsonPath).String()
+	if defaultGw == "" {
+		msg := "default gw missing"
+		defaultGwLog := log.WithValues("path", defaultGwGjsonPath)
+		defaultGwLogDebug := defaultGwLog.V(1)
+		if defaultGwLogDebug.Enabled() {
+			defaultGwLogDebug.Info(msg, "state", gjsonCurrentState.String())
+		} else {
+			defaultGwLog.Info(msg)
 		}
-
-		return true, nil
-	})
+		return "", nil
+	}
+	return defaultGw, nil
 }
 
-func runPing(_ client.Client, timeout time.Duration) error {
+func runPing(_ client.Client) error {
 	defaultGw, err := defaultGw()
 	if err != nil {
 		return errors.Wrap(err, "failed to retrieve default gw at runProbes")
 	}
 
-	pingOutput, err := ping(defaultGw, timeout)
+	pingOutput, err := ping(defaultGw)
 	if err != nil {
 		return errors.Wrapf(err, "error pinging default gateway -> output: %s", pingOutput)
 	}
 	return nil
 }
 
-func runDNS(_ client.Client, timeout time.Duration) error {
+func runDNS(_ client.Client) error {
 	runningServersGJsonPath := "dns-resolver.running.server"
 	errs := []error{}
-	runningNameServers := []gjson.Result{}
 
-	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
-		// Get the name servers at node since the ones at container are not accurate
-		currentStateAsGJson, err := currentStateAsGJson()
-		if err != nil {
-			return false, errors.Wrap(err, "failed retrieving current state to get name resolving config")
-		}
-		runningNameServers = currentStateAsGJson.Get(runningServersGJsonPath).Array()
-		if len(runningNameServers) == 0 {
-			return false, fmt.Errorf("missing name servers at '%s' on %s", runningServersGJsonPath, currentStateAsGJson.String())
+	// Get the name servers at node since the ones at container may not be up to date
+	currentStateAsGJson, err := currentStateAsGJson()
+	if err != nil {
+		return errors.Wrap(err, "failed retrieving current state to get name resolving config")
+	}
+	runningNameServers := currentStateAsGJson.Get(runningServersGJsonPath).Array()
+	if len(runningNameServers) == 0 {
+		return fmt.Errorf("missing name servers at '%s' on %s", runningServersGJsonPath, currentStateAsGJson.String())
+	}
+	for _, runningNameServer := range runningNameServers {
+		r := &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{
+					Timeout: defaultDNSProbeTimeout,
```

Comment:
Do we need this Timeout ? we are going to poll

Reply:
Removed